ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -221,6 +235,15 @@ public void testFindMissingKeys() throws Exception {
     Set<StoreKey> missingKeys = store.findMissingKeys(keys);
     verify(dest).getBlobMetadata(anyList());
     assertEquals("Wrong number of missing keys", count, missingKeys.size());
+
+    // Add keys to cache and rerun (should be cached)
+    for (StoreKey storeKey : keys) {
+      store.addToCache(storeKey.getID(), CloudBlobStore.BlobState.CREATED);
+    }
+    missingKeys = store.findMissingKeys(keys);
+    assertTrue("Expected no missing keys", missingKeys.isEmpty());
+    // getBlobMetadata should not have been called a second time.
+    verify(dest).getBlobMetadata(anyList());
```

Comment:
Perhaps a unit test to confirm that the LRU nature of the cache works as expected (set the cache limit artificially low, add PUTs over that number, confirm that redoing the PUTs yields (number of PUTs - limit) unskipped cloud PUTs)?

Reply:
Done (method a bit different).